test/puppetlabs/trapperkeeper/services/authorization/authorization_service_test.clj

```diff
@@ -451,3 +451,66 @@
             (is (not (-> (get-service app :PlumbingService)
                          (call-authorization-check catalog-request-nocert)
                          (rules/authorized?))))))))))
+
+(deftest ^:integration subject-alt-names
+  (testing "Matching subjectAlternativeName"
+    (let [allowable-alt-names ["testing.domain.org" "foo.domain.org" "domain.foo.org" "test.domain.com"]
+          deniable-alt-names ["testing.bad.org" "foo.bad.org" "bad.foo.org" "test.bad.com"]
+          allowable-ext (ssl-utils/subject-dns-alt-names allowable-alt-names false)
+          deniable-ext (ssl-utils/subject-dns-alt-names deniable-alt-names false)
+          allowable-cert (testutils/create-certificate "test.domain.org"
+                                                       [allowable-ext])
+          deniable-cert (testutils/create-certificate "test.bad.org"
+                                                      [deniable-ext])
+          ext-rules (conj basic-rules
+                          {:match-request
+                           {:path "/puppet/v4/catalog"
+                            :type "path"
+                            :method "get"}
+                           :deny {:extensions {:subjectAlternativeName ["testing.bad.org" "bad.foo.org"]}}
+                           :allow {:extensions {:subjectAlternativeName ["foo.domain.org" "test.domain.com"]}}
+                           :sort-order 100
+                           :name "puppetlabs v4 catalog"})
+          certname-rules (conj basic-rules
+                               {:match-request
+                                {:path "/puppet/v4/catalog"
+                                 :type "path"
+                                 :method "get"}
+                                :deny {:certname "foo.bad.org"}
+                                :allow {:certname "testing.domain.org"}
+                                :sort-order 100
+                                :name "puppetlabs v4 catalog"})
+          auth-opts {:oid-map {}}]
+      (testing "as an extension"
+        (with-test-logging
+          (with-app-with-config
+            app
+            [plumbing-service authorization-service]
+            (assoc-in minimal-config [:authorization :rules] ext-rules)
+            (testing "with allow rule"
+              (let [req (testutils/request "/puppet/v4/catalog" :get "127.0.0.1" allowable-cert)]
+                (is (-> (get-service app :PlumbingService)
+                        (call-authorization-check req auth-opts)
+                        rules/authorized?))))
+            (testing "with deny rule"
+              (let [req (testutils/request "/puppet/v4/catalog" :get "127.0.0.1" deniable-cert)]
+                (is (not (-> (get-service app :PlumbingService)
+                             (call-authorization-check req auth-opts)
+                             rules/authorized?))))))))
+
+      (testing "as a certname match"
```

Comment:
These tests look good.  I think it might be good also to have at least one more flavor of domain rule - maybe a regex? - to show that the other domain rule type still matches a subjaltname from the certificate as expected?